clap_man/src/render.rs

```diff
@@ -30,8 +30,38 @@ pub(crate) fn description(roff: &mut Roff, app: &clap::App) {
 
 pub(crate) fn synopsis(roff: &mut Roff, app: &clap::App) {
     let mut line = vec![bold(app.get_name()), roman(" ")];
+    let mut items: Vec<_> = app
+        .get_arguments()
+        .filter(|i| !i.is_set(ArgSettings::Hidden))
+        .collect();
+
+    // Render the help argument
+    items
+        .iter()
+        .filter(|a| a.get_long().unwrap_or(&"default".to_string()) == "help")
+        .for_each(|opt| {
+            let (lhs, rhs) = option_markers(opt);
+            let long = opt.get_long().unwrap();
+            let short = opt.get_short().unwrap();
+            line.push(roman(lhs));
+            line.push(bold(&format!("-{}", short)));
+            line.push(roman("|"));
+            line.push(bold(&format!("--{}", long)));
+            line.push(roman(rhs));
+            line.push(roman(" "));
+        });
+
+    // Sort the remaining arguments based on display order
+    items.sort_by(|a, b| {
+        a.get_display_order()
+            .partial_cmp(&b.get_display_order())
+            .unwrap()
+    });
```

Comment:
We should only be sorting when `DeriveDisplayOrdet` is set

Reply:
Correction: we shouldn't need to check `DeriveDisplayOrdet` but we a smarter key than just `get_display_order()`.

In `--help`, the logic is roughly
```rust
items.sort_by_key(|arg| {
            let subkey = if let Some(x) = arg.short {
                let mut s = x.to_ascii_lowercase().to_string();
                s.push(if x.is_ascii_lowercase() { '0' } else { '1' });
                s
            } else if let Some(x) = arg.long {
                x.to_string()
            } else {
                let mut s = '{'.to_string();
                s.push_str(arg.name);
                s
            };

    (key.get_display_order(), subkey);
});
```

Reply:
This was very helpful, I utilised this idea in the updated version.